hypershift-operator/controllers/util/util.go

```diff
@@ -0,0 +1,39 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func ParseNamespacedName(name string) types.NamespacedName {
```

Comment:
This is really confusing and I don't think we need it. In both cases where it's used we have access to an object with name and namespace through typed fields.

Reply:
not really right? this parses the parent name included as a string in an annotation for the watched child CR

Reply:
Uhhh, yeah. Right. I guess that means it's even more confusing than I thought...

How about instead a function like:

```go
func getParentHostedClusterName(obj ctrlclient.Object) (types.NamespacedName, error) {
	var hostedClusterName string
	if obj.GetAnnotations() != nil {
		hostedClusterName = obj.GetAnnotations()[hostedClusterAnnotation]
	}
	if hostedClusterName == "" {
		return types.NamespacedName{}, fmt.Errorf("no reference found")
	}
	return hostedClusterName, nil
}
```

Reply:
It's not just name, it's also namespace

Reply:
Added a comment on `ParseNamespacedName` to reduce confusion.

Reply:
Thanks!